### ISSUE TITLE
data-router performance drop in 0.2.0

### ISSUE THREAD

**Issue by damszew** [epiphany-platform/CommonDataLayer#215]
**Describe the bug**
After updating data-router from one build on 852db19 to released 0.2.0, I am experiencing quite big performance drop (from 200 msg/s to around 160 msg/s).

**Expected behavior**
Similar performance between versions.

**To Reproduce**
1. Setup environment with kafka mq
2. Send around 200 msg/s to be stored in VictoriaMetrics
3. Observe data-router metrics

**Additional context**

**Comment by frondeus** [comment]
Well, I investigated a bit and found that between these two versions in data-router we changed only these two things:
* Added dynamic message queue selection. Each time we take the next value from the topic/queue we `match` if this is Kafka or AMQP.
In theory, it should be simple integer comparison.
* Added order group id - and that one might cause performance drop not only because there is additional field but also we do string replacement from `-` to `.`.

One question tho, did you check on the release target?
